### PR TITLE
Add UI features and algorithm to support label inference

### PR DESCRIPTION
**Pull request by GabrielKS**
+ Phone now receives and processes label inference data structure from the server to determine what inferences to show the user
+ Makes labels red if they have no value, yellow if they are inferred, and green if they are user input
+ Adds a "verify" button that turns yellow labels green

### CODE REVIEW

**Comment by shankari** [comment]
@GabrielKS can you add some screenshots or animated gifs here to make it easier to review without pulling the changes and running them?

File: www/templates/diary/infinite_scroll_list.html
Comment (shankari) [inline www/templates/diary/infinite_scroll_list.html:69]: FYI, please continue using properties instead of functions here. Using functions to retrieve the `ng-class`, although it works, plays havoc with the caching and the performance really suffers.

**Review by shankari — APPROVED** [review]
Looks pretty straightforward; I will merge after you upload some screenshots/animated gifs

**Comment by GabrielKS** [comment]
Below is a GIF that showcases all of the changes implemented. The inference data is made up for instructional purposes.
Showcase GIF

Here's what is happening for each trip:
* Trip 1: No inference possible. Same behavior as before these changes, except blank labels are red instead of gray.
* Trip 2: Both labels can be inferred. Verify button marks labels as user-confirmed, turning them green.
* Trip 3: Only mode can be inferred; verify button confirms it.
* Trip 4: Displayed inferences are incorrect. User inputs mode, at which point the system can correctly infer purpose.
* Trip 5: Only mode can be inferred with enough confidence to display it; once user confirms that, confidence in the purpose inference increases to the point that it can be displayed.
* Trip 6: Only mode can be inferred with enough confidence to display it, but it is incorrect; user enters purpose, at which point the system can correctly infer mode.

**Comment by shankari** [comment]
Please submit the PR to fix the confirmation check for Trip 1 after I merge this.

**Pull request merged**
